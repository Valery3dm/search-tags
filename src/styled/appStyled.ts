import styled from 'styled-components';

const AppStyled: string = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export default AppStyled;
