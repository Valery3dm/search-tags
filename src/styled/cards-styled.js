import Card from "react-bootstrap/Card";
import styled from "styled-components";

const ListCards = styled.ul`
  margin-top: 4rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  list-style-type: none;
  padding: 0;
`;

const CardStyled = styled(Card)`
  width: 18rem;
  margin: 3px;
  height: 18rem;
`;
const CardImgStyled = styled(Card.Img)`
  height: 12rem;
  object-fit: cover;
`;

export { ListCards, CardStyled, CardImgStyled };
