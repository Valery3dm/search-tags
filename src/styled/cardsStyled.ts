import Card from 'react-bootstrap/Card';
import styled from 'styled-components';

const ListCards: any = styled.ul`
  margin-top: 4rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  list-style-type: none;
  padding: 0;
`;

const CardStyled: any = styled(Card)`
  width: 18rem;
  margin: 5px;
  height: 18rem;
  box-shadow: 0 0 10px rgb(0 0 0 / 50%);
`;

const CardImgStyled: any = styled(Card.Img)`
  height: 12rem;
  object-fit: cover;
`;

export { ListCards, CardStyled, CardImgStyled };
