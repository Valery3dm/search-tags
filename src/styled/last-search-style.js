import styled from "styled-components";
import Button from "react-bootstrap/Button";

const ButtonStyled = styled(Button)`
  margin: 4px 2px 0px 2px;
`;

const LastItemSearch = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 0px 10px 0px 10px;
`;

const LastThreeItemSearch = styled.ul`
  display: flex;
  list-style-type: none;
  padding-left: 0px;
`
export { LastItemSearch, ButtonStyled, LastThreeItemSearch };
