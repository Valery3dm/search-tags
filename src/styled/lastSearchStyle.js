import styled from 'styled-components';
import Button from 'react-bootstrap/Button';

const ButtonStyled = styled(Button)`
  margin: 10px 4px 0px 4px;
  box-shadow: 0 0 10px rgb(0 0 0 / 50%);
`;

const LastItemSearch = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 0px 10px 0px 10px;
`;

const LastThreeItemSearch = styled.ul`
  display: flex;
  list-style-type: none;
  padding-left: 0px;
`

export { LastItemSearch, ButtonStyled, LastThreeItemSearch };
