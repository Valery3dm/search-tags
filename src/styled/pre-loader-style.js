import styled, { keyframes } from "styled-components";

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Rotate = styled.div`
  display: inline-block;
  animation: ${rotate} 2s linear infinite;
  padding: 3rem 1rem;
  font-size: 3rem;
`;

export default Rotate;
