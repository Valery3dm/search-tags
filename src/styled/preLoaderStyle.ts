import styled, { keyframes } from 'styled-components';

const rotate: any = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Rotate: string = styled.div`
  display: inline-block;
  animation: ${rotate} 2s linear infinite;
  padding: 3rem 1rem;
  font-size: 3rem;
`;

export default Rotate;
