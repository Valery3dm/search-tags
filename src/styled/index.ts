import AppStyled from './appStyled';
import { ListCards, CardStyled, CardImgStyled } from './cardsStyled';
import { LastItemSearch, ButtonStyled, LastThreeItemSearch } from './lastSearchStyle';
import Rotate from './preLoaderStyle';
import SearchPanelStyle from './searchPanelStyle';

export {
    AppStyled,
    ListCards,
    CardStyled,
    CardImgStyled,
    LastItemSearch,
    ButtonStyled,
    LastThreeItemSearch,
    Rotate,
    SearchPanelStyle
};
