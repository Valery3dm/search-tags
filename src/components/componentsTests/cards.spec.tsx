import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import Card from 'react-bootstrap/Card';

import { Cards } from '../';

import { CardImgStyled, CardStyled, ListCards } from '../../styled';

Enzyme.configure({ adapter: new Adapter() });

describe('Cards component', () => {
    const itemsList = [
        {
            id: 1,
            largeImageURL: 'qweasdzxc123',
            tags: 'item1'
        },
        {
            id: 2,
            largeImageURL: 'asdzxc123',
            tags: 'item2'
        },
        {
            id: 3,
            largeImageURL: 'zxc123',
            tags: 'item3'
        }
    ];
    
    const component = shallow(
        <Cards itemsList={itemsList}/>
    );

    it('Cards Snapshot component', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

    it('find ListCards', () => {
        expect(component.find(ListCards)).toHaveLength(1);
    });

    it('find CardStyled', () => {
        expect(component.find(CardStyled)).toHaveLength(3);
    });

    it('find CardImgStyled', () => {
        expect(component.find(CardImgStyled)).toHaveLength(3);
    });

    it('find Card.Body', () => {
        expect(component.find(Card.Body)).toHaveLength(3);
    });

    it('find Card.Title', () => {
        expect(component.find(Card.Title)).toHaveLength(3);
    });
});
