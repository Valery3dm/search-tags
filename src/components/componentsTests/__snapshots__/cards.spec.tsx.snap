// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component Cards Snapshot component 1`] = `
<styled.ul>
  <li
    key="1"
  >
    <Styled(Card)>
      <Styled(CardImg)
        src="qweasdzxc123"
        variant="top"
      />
      <CardBody>
        <CardTitle>
          item1
        </CardTitle>
      </CardBody>
    </Styled(Card)>
  </li>
  <li
    key="2"
  >
    <Styled(Card)>
      <Styled(CardImg)
        src="asdzxc123"
        variant="top"
      />
      <CardBody>
        <CardTitle>
          item2
        </CardTitle>
      </CardBody>
    </Styled(Card)>
  </li>
  <li
    key="3"
  >
    <Styled(Card)>
      <Styled(CardImg)
        src="zxc123"
        variant="top"
      />
      <CardBody>
        <CardTitle>
          item3
        </CardTitle>
      </CardBody>
    </Styled(Card)>
  </li>
</styled.ul>
`;
